tools/testsys/src/aws_resources.rs

```diff
@@ -342,6 +341,295 @@ impl Migration for AwsK8s {
     }
 }
 
+pub(crate) struct AwsEcs {
+    pub(crate) arch: String,
+    pub(crate) variant: String,
+    pub(crate) region: String,
+    pub(crate) assume_role: Option<String>,
+    pub(crate) instance_type: Option<String>,
+    pub(crate) ami: String,
+    pub(crate) secrets: Option<BTreeMap<String, SecretName>>,
+    pub(crate) target_cluster_name: Option<String>,
+    pub(crate) tuf_repo: Option<TufRepoConfig>,
+    pub(crate) starting_version: Option<String>,
+    pub(crate) starting_image_id: Option<String>,
+    pub(crate) migrate_starting_commit: Option<String>,
+    pub(crate) migrate_to_version: Option<String>,
+    pub(crate) capabilities: Option<Vec<String>>,
+}
+
+impl AwsEcs {
+    /// Create the necessary test and resource crds for the specified test type.
+    pub(crate) async fn create_crds(
+        &self,
+        test: TestType,
+        testsys_images: &TestsysImages,
+    ) -> Result<Vec<Crd>> {
+        match test {
+            TestType::Conformance => {
+                return Err(anyhow!(
+                    "Conformance testing for ECS variants is not supported."
+                ))
+            }
+            TestType::Quick => self.ecs_test_crds(testsys_images),
+            TestType::Migration => self.migration_test_crds(testsys_images).await,
+        }
+    }
+
+    fn ecs_test_crds(&self, testsys_images: &TestsysImages) -> Result<Vec<Crd>> {
+        let crds = vec![
+            self.ecs_crd(testsys_images)?,
+            self.ec2_crd(testsys_images, None)?,
+            self.ecs_test_crd("-test", None, testsys_images)?,
+        ];
+        Ok(crds)
+    }
+
+    async fn migration_test_crds(&self, testsys_images: &TestsysImages) -> Result<Vec<Crd>> {
+        let ami = self
+            .starting_image_id
+            .as_ref()
+            .unwrap_or(
+                &get_ami_id(
+                    format!(
+                        "bottlerocket-{}-{}-{}-{}",
+                        self.variant, self.arch, self.starting_version.as_ref().context("The starting version must be provided for migration testing")?, self.migrate_starting_commit.as_ref().context("The commit for the starting version must be provided if the starting image id is not")?
+                    ), & self.arch,
```

Comment:
The `AwsEcs` struct definition is confusing. Why are `starting_version` and `migrate_starting_commit` optional if both are "required" here and in the implementation of `Migration` for the struct?

Reply:
That is because `starting_version` and `migrate_starting_commit` are not used for quick testing. `cargo make -e TESTSYS_TEST=quick -e BUILDSYS_VARIANT="aws-ecs-1" test`